### PR TITLE
Web bootstrap markdown 80

### PR DESCRIPTION
**Pull request by njeudy**
[web_bootstrap_markdown] Initial release for new text form widget with markdown syntaxe.

First inital release.

### CODE REVIEW

**Comment by njeudy** [comment]
someone knows how to add images on README.rst that can be view in odoo module description ?

**Comment by pedrobaeza** [comment]
Interesting widget! I would call it `web_widget_text_markdown`, not emphasizing in the background technology (you could mention it anyway on the description), but more on the type for the widget (text), the extra option (markdown), and the word widget for indicating that it is indeed a widget.

**Comment by njeudy** [comment]
Ok, you are right. I just rename this module and convert `web_bootstrap_markdown` to `web_widget_text_markdown`

I will improve and add markdown for odoo syntax to support for exemple a link for account_invoice, project.task .. etc ..

Is it possible to use issue on this repository to store improvement and wishes ?

**Comment by pedrobaeza** [comment]
Yeah, an issue can be a wishlist request if you want, or to track a progress of a task. We have used it in other occassions for this purpose.

**Comment by njeudy** [comment]
How can I do that, I can not add label on issue in this project  ?

**Comment by pedrobaeza** [comment]
Create an issue, write the number of the PR with a `#` preceding it in the text of the issue, and a reference will be automatically created.

**Comment by njeudy** [comment]
Ok I will do that for the next release. This widget is already usable and permit to have good task description for example with external images :)

File: web_widget_text_markdown/README.rst [hunk 1]
```diff
@@ -0,0 +1,44 @@
+==============================
+Add new text field form widget
+==============================
+
+Description
+-----------
+
+This modules add a new widget for text field in form view on Odoo:
+
+- In readonly mode, it use marked to parse and render to html markdown syntaxe.
```
Comment (pedrobaeza) [inline web_widget_text_markdown/README.rst]: s/it use marked/it uses marks

Comment (pedrobaeza) [inline web_widget_text_markdown/README.rst]: s/syntaxe/syntax

File: web_widget_text_markdown/README.rst [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+Was tested on openerp trunk, 8.0 branch.
+
+Example
+-------
+
+Your XML form view definition should contain::
+
+    ...
+    <field name="description" widget="bootstrap_markdown"/>
```
Comment (pedrobaeza) [inline web_widget_text_markdown/README.rst]: s/description/field_name (more generic)

File: web_widget_text_markdown/README.rst [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    ...
+
+Note / Todo
+-----------
+
+- Improve user experience with Odoo specific syntaxe
```
Comment (pedrobaeza) [inline web_widget_text_markdown/README.rst]: s/syntaxe/syntax

File: web_widget_text_markdown/README.rst [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+- Improve user experience with Github specific syntaxe
```
Comment (pedrobaeza) [inline web_widget_text_markdown/README.rst]: s/syntaxe/syntax

File: web_widget_text_markdown/__openerp__.py [hunk 5]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (pedrobaeza) [inline web_widget_text_markdown/__openerp__.py:21]: Put AGPL disclaimer

**Comment by pedrobaeza** [comment]
Some minor issues, the rest :+1:

**Comment by pedrobaeza** [comment]
:+1:

**Comment by njeudy** [comment]
Sorry for the initial release, when testing today, I see that some of file I copied to OCA web project where no here .. This version is working again.

**Comment by njeudy** [comment]
I just rebase on latest 8.0 version (don't know if I should in fact ...)

**Comment by yvaucher** [comment]
@njeudy rebasing is good practice IMO with git. I find it cleaner than merging in term of history.

From a technical point of view :+1: 

But I wonder if it is possible to include all those js lib with different licenses. Are these compatible ?

**Comment by njeudy** [comment]
@yvaucher : For js lib licenses I don't know. They are opensources .. What is the best way to use external js lib ? Odoo use many  of them, have they all the same licence ?

Thanks for your comment,

**Comment by yvaucher** [comment]
@njeudy It was a curiosity question I have no idea how they should work together.

**Comment by guewen** [comment]
The libraries are licensed under Apache 2.0, MIT and BSD licenses, AFAIK there is no compatibility problems with AGPLv3. However, `marked` is MIT licensed and must be redistributed with the permission notice, could you add this file https://github.com/chjj/marked/blob/master/LICENSE in the `marked` directory?

**Comment by bealdav** [comment]
bootstrap-markdown is really impressive. Really good news to have this module merged next. Thanks for this work.

**Comment by njeudy** [comment]
Ok I add the license file for marked. someone can say to me how I can add test for this kind of module ?
Il think there is a lots of more options to add but it is ok for a first release.

**Comment by njeudy** [comment]
I just rebase on latest oca/8.0 branch

**Comment by guewen** [comment]
Thanks! :+1:

**Pull request merged**